### ISSUE TITLE
Vulnerability in `webpki` 0.22.0

### ISSUE THREAD

**Issue by thibault-martinez** [bytebeamio/rumqtt#689]
The vulnerability https://rustsec.org/advisories/RUSTSEC-2023-0052 has been issued yesterday and the crate is present in your dependency tree, making our CI to fail.

Updating your dependency on `async-tungstenite` from `0.22` to `0.23` should fix the issue.

Would you be so kind as to update it and release a new version please 🙏🏻 ?

Thanks in advance.

**Comment by swanandx** [comment]
Hey, thanks for reporting! regarding the CVE, we updated `rustls-webpki` ( #687 ) .
For updating `async-tungstenite` to `0.23`, it seems some other dependency, specifically `ws_stream_tungstenite`, has pinned it to `0.22` which is causing conflicts, leading CI to fail in #690 . 
I don't think `async-tungstenite` depends no `rustls-webpki`, so we might not need to update it! wdyt?

**Comment by thibault-martinez** [comment]
Hello,

`async-tungstenite 0.22.2` transitively relies on `webpki 0.22.0` which has the CVE.

```
Crate:     webpki
Version:   0.22.0
Title:     webpki: CPU denial of service in certificate path building
Date:      2023-08-22
ID:        RUSTSEC-2023-0052
URL:       https://rustsec.org/advisories/RUSTSEC-2023-0052
Severity:  7.5 (high)
Solution:  No fixed upgrade is available!
Dependency tree:
webpki 0.22.0
└── tungstenite 0.19.0
    ├── ws_stream_tungstenite 0.10.0
    │   ├── rumqttd 0.17.0
    │   │   └── benchmarks 0.4.0
    │   └── rumqttc 0.22.0
    │       └── benchmarks 0.4.0
    └── async-tungstenite 0.22.2
        ├── ws_stream_tungstenite 0.10.0
        ├── rumqttd 0.17.0
        └── rumqttc 0.22.0
```

**Comment by thibault-martinez** [comment]
Hello, thank you for fixing it. Are you planning on releasing a new version?

**Comment by swanandx** [comment]
yup, I was planning to close this issue after we make a release, but mentioning it in PR closed this beforehand haha. Both rumqttc as well as rumqttd will be released very soon.

**Comment by thibault-martinez** [comment]
Thank you very much. Have a good day.

**Comment by swanandx** [comment]
Same to you, have a great day. Also, Thanks for reporting the vulnerability, much appreciated!

**Comment by swanandx** [comment]
Hey @thibault-martinez , webpki released patch version 0.22.1 with the fix! So we don't need to update the deps, you can just run `cargo update` and then check `cargo audit`. That means, you won't have to wait for next release, rumqttc 0.22.0 will work just fine :100: 

Let me know if it works for you!

**Comment by thibault-martinez** [comment]
Updated, thanks for the ping 🙏🏻
